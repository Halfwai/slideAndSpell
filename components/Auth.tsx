import React, { useState, useEffect } from 'react'
import { Alert, StyleSheet, View, Image, TouchableOpacity, Dimensions } from 'react-native'

import { Button, Input } from '@rneui/themed'

import { useFonts } from 'expo-font';

import { COLOURS } from '@/constants/colours'

import Welcome from '@/components/Welcome'
import SignIn from '@/components/SignIn';
import SignUp from '@/components/SignUp';

export default function Auth() {


    const [currentMenu, setCurrentMenu] = useState('welcome')
    const [slideIn, setSlideIn] = useState(false);
    const [welcomePos, setWelcomePos] = useState({x: 0, y: 0})
    const [signInPos, setSignInPos] = useState({x: - Dimensions.get('window').width, y: 0})
    const [signUpPos, setSignUpPos] = useState({x: Dimensions.get('window').width * 2, y: 0})

    useEffect(() => {
        if (currentMenu === 'welcome') {
            setWelcomePos({x: 0, y: 0})
        } else {
            setWelcomePos({x: 0, y: -Dimensions.get('window').height})
        }
        if (currentMenu === 'signIn') {
            setSignInPos({x: 0, y: 0})
        } else {
            setSignInPos({x: -Dimensions.get('window').width, y: 0})
        }
        if (currentMenu === 'signUp') {
            setSignUpPos({x: 0, y: 0})
        } else {
            setSignUpPos({x: Dimensions.get('window').width * 2, y: 0})
        }
    }, [currentMenu])

    // async function signInWithEmail() {
    //     setLoading(true)
    //     const { error } = await supabase.auth.signInWithPassword({
    //         email: email,
    //         password: password,
    //     })

    //     if (error) Alert.alert(error.message)
    //     setLoading(false)
    // }

    // async function signUpWithEmail() {
    //     setLoading(true)
    //     const {
    //         data: { session },
    //         error,
    //     } = await supabase.auth.signUp({
    //         email: email,
    //         password: password,
    //         options: {
    //             data: {
    //                 display_name: 'username'
    //             }
    //         }
    //     })

    //     if (error) Alert.alert(error.message)
    //     if (!session) Alert.alert('Please check your inbox for email verification!')
    //     setLoading(false)
    // }

    return (
        <View style={styles.container}>
            <View style={styles.imageContainer}>
                <Image 
                    source={require("@/assets/images/logo.png")}
                    style={{ width: 300, height: 180, zIndex: 1 }}
                    resizeMode="contain"
                />
            </View>

            <View style={{flex: 1, width: '100%'}}>
                <Welcome 
                    position={welcomePos}
                    setMenu={(state : string) => setCurrentMenu(state)}
                    setSlideIn={(state : boolean) => setSlideIn(state)}
                />
                <SignIn 
                    position={signInPos}
                    setMenu={(state : string) => setCurrentMenu(state)}
                />
                <SignUp 
                    position={signUpPos}
                    setMenu={(state : string) => setCurrentMenu(state)}
                />
            </View>


        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        backgroundColor: COLOURS.blue,
        height: '100%',
        alignItems: 'center',
        justifyContent: 'flex-start',
    },
    imageContainer: {
        width: '100%',
        alignItems: 'center',
        paddingTop: 60,
        zIndex: 1,
        backgroundColor: COLOURS.blue,
    },
})


{/* <View style={[styles.verticallySpaced]}>
<Input
    leftIcon={{ type: 'font-awesome', name: 'envelope' }}
    onChangeText={(text) => setEmail(text)}
    value={email}
    placeholder="email@address.com"
    autoCapitalize={'none'}
/>
</View>
<View style={styles.verticallySpaced}>
<Input
    leftIcon={{ type: 'font-awesome', name: 'lock' }}
    onChangeText={(text) => setPassword(text)}
    value={password}
    secureTextEntry={true}
    placeholder="Password"
    autoCapitalize={'none'}
    style={{backgroundColor: "white"}}
/>
</View>
<View style={[styles.verticallySpaced]}>
<Button title="Sign in" disabled={loading} onPress={() => signInWithEmail()} />
</View>
<View style={styles.verticallySpaced}>
<Button title="Sign up" disabled={loading} onPress={() => signUpWithEmail()} />
</View> */}