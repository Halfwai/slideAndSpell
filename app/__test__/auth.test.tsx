import { render } from '@testing-library/react-native';

import Auth from '@/app/auth';

afterEach(() => {
    jest.clearAllMocks();  // Clears all Jest mocks
    jest.useRealTimers();  // Resets timers
});

it(`Auth Component renders correctly`, () => {
    const { toJSON } = render(
        <Auth/>
    );
    expect(toJSON()).toMatchSnapshot();
});