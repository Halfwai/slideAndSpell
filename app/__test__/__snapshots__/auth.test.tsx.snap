// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Component renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#CBE6F7",
      "height": "100%",
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#CBE6F7",
        "height": 400.2,
        "width": "100%",
        "zIndex": 1,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        {
          "testUri": "../../../assets/images/logo.png",
        }
      }
      style={
        {
          "height": "100%",
          "width": "90%",
          "zIndex": 1,
        }
      }
    />
  </View>
  <View
    style={
      {
        "flex": 1,
        "width": "100%",
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#CBE6F7",
          "position": "absolute",
          "transform": [
            {
              "translateX": 0,
            },
            {
              "translateY": 0,
            },
          ],
          "width": "100%",
          "zIndex": 0,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "borderRadius": 10,
            "padding": 12,
            "width": "90%",
          }
        }
      >
        <Text
          style={
            [
              {
                "fontFamily": "Quicksand",
              },
              {
                "color": "black",
                "fontSize": 30,
                "fontWeight": "bold",
              },
            ]
          }
        >
          Hi
        </Text>
        <Text
          style={
            [
              {
                "fontFamily": "Quicksand",
              },
              {
                "color": "black",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 20,
                "textAlign": "center",
              },
            ]
          }
        >
          Welcome to Slide and Spell, please sign in for stats and leaderboards, or hit endless mode to play without stats or tracking.
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "space-between",
            "width": "100%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#8BF0BC",
              "borderRadius": 10,
              "borderWidth": 1,
              "marginBottom": 20,
              "opacity": 1,
              "padding": 12,
              "width": "90%",
            }
          }
          testID="authButton"
        >
          <Text
            style={
              [
                {
                  "fontFamily": "Quicksand",
                },
                {
                  "color": "black",
                  "fontSize": 20,
                  "fontWeight": "bold",
                },
              ]
            }
          >
            Sign In
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderColor": "#8BF0BC",
              "borderRadius": 10,
              "borderWidth": 1,
              "marginBottom": 20,
              "opacity": 1,
              "padding": 12,
              "width": "90%",
            }
          }
          testID="authButton"
        >
          <Text
            style={
              [
                {
                  "fontFamily": "Quicksand",
                },
                {
                  "color": "black",
                  "fontSize": 20,
                  "fontWeight": "bold",
                },
              ]
            }
          >
            Sign Up
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "height": "60%",
            "justifyContent": "flex-end",
            "width": "100%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderColor": "#8BF0BC",
              "borderRadius": 10,
              "borderWidth": 1,
              "marginBottom": 20,
              "opacity": 1,
              "padding": 12,
              "width": "90%",
            }
          }
          testID="authButton"
        >
          <Text
            style={
              [
                {
                  "fontFamily": "Quicksand",
                },
                {
                  "color": "black",
                  "fontSize": 20,
                  "fontWeight": "bold",
                },
              ]
            }
          >
            Endless Mode
          </Text>
        </View>
      </View>
    </View>
    <View
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#CBE6F7",
          "position": "absolute",
          "transform": [
            {
              "translateX": -750,
            },
            {
              "translateY": 0,
            },
          ],
          "width": "100%",
          "zIndex": 0,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            [
              {
                "fontFamily": "Quicksand",
              },
              {
                "fontSize": 30,
              },
            ]
          }
        >
          Sign In
        </Text>
        <View
          style={
            {
              "alignItems": "center",
              "marginBottom": 20,
              "width": "100%",
            }
          }
        >
          <TextInput
            autoComplete="email"
            autoCorrect={false}
            inputMode="email"
            onChangeText={[Function]}
            placeholder="Email"
            style={
              {
                "backgroundColor": "white",
                "borderRadius": 10,
                "borderWidth": 1,
                "fontSize": 20,
                "height": 50,
                "margin": 10,
                "padding": 10,
                "width": "90%",
              }
            }
            value=""
          />
          <TextInput
            autoCorrect={false}
            onChangeText={[Function]}
            placeholder="Password"
            secureTextEntry={true}
            style={
              {
                "backgroundColor": "white",
                "borderRadius": 10,
                "borderWidth": 1,
                "fontSize": 20,
                "height": 50,
                "margin": 10,
                "padding": 10,
                "width": "90%",
              }
            }
            value=""
          />
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#8BF0BC",
              "borderRadius": 10,
              "borderWidth": 1,
              "marginBottom": 20,
              "opacity": 1,
              "padding": 12,
              "width": "90%",
            }
          }
          testID="authButton"
        >
          <Text
            style={
              [
                {
                  "fontFamily": "Quicksand",
                },
                {
                  "color": "black",
                  "fontSize": 20,
                  "fontWeight": "bold",
                },
              ]
            }
          >
            Sign In
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderColor": "#8BF0BC",
              "borderRadius": 10,
              "borderWidth": 1,
              "marginBottom": 20,
              "opacity": 1,
              "padding": 12,
              "width": "90%",
            }
          }
          testID="authButton"
        >
          <Text
            style={
              [
                {
                  "fontFamily": "Quicksand",
                },
                {
                  "color": "black",
                  "fontSize": 20,
                  "fontWeight": "bold",
                },
              ]
            }
          >
            Back
          </Text>
        </View>
      </View>
    </View>
    <View
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#CBE6F7",
          "position": "absolute",
          "transform": [
            {
              "translateX": 1500,
            },
            {
              "translateY": 0,
            },
          ],
          "width": "100%",
          "zIndex": 0,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            [
              {
                "fontFamily": "Quicksand",
              },
              {
                "fontSize": 30,
              },
            ]
          }
        >
          Sign Up
        </Text>
        <View
          style={
            {
              "alignItems": "center",
              "width": "100%",
            }
          }
        >
          <TextInput
            autoComplete="email"
            autoCorrect={false}
            inputMode="email"
            onChangeText={[Function]}
            placeholder="Email"
            style={
              {
                "backgroundColor": "white",
                "borderRadius": 10,
                "borderWidth": 1,
                "fontSize": 20,
                "height": 50,
                "marginTop": 10,
                "padding": 10,
                "width": "90%",
              }
            }
            value=""
          />
          <TextInput
            autoCorrect={false}
            onChangeText={[Function]}
            placeholder="Display Name"
            style={
              {
                "backgroundColor": "white",
                "borderRadius": 10,
                "borderWidth": 1,
                "fontSize": 20,
                "height": 50,
                "marginTop": 10,
                "padding": 10,
                "width": "90%",
              }
            }
            value=""
          />
          <TextInput
            autoCorrect={false}
            onChangeText={[Function]}
            placeholder="Password"
            secureTextEntry={true}
            style={
              {
                "backgroundColor": "white",
                "borderRadius": 10,
                "borderWidth": 1,
                "fontSize": 20,
                "height": 50,
                "marginTop": 10,
                "padding": 10,
                "width": "90%",
              }
            }
            value=""
          />
          <TextInput
            autoCorrect={false}
            onChangeText={[Function]}
            placeholder="Confirm Password"
            secureTextEntry={true}
            style={
              {
                "backgroundColor": "white",
                "borderRadius": 10,
                "borderWidth": 1,
                "fontSize": 20,
                "height": 50,
                "marginTop": 10,
                "padding": 10,
                "width": "90%",
              }
            }
            value=""
          />
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              [
                {
                  "fontFamily": "Quicksand",
                },
                undefined,
              ]
            }
          >
            By Signing up you agree to our 
            <Text
              style={
                [
                  {
                    "fontFamily": "Quicksand",
                  },
                  {
                    "textDecorationLine": "underline",
                  },
                ]
              }
            >
              Privacy Policy
            </Text>
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#8BF0BC",
              "borderRadius": 10,
              "borderWidth": 1,
              "marginBottom": 20,
              "opacity": 1,
              "padding": 12,
              "width": "90%",
            }
          }
          testID="authButton"
        >
          <Text
            style={
              [
                {
                  "fontFamily": "Quicksand",
                },
                {
                  "color": "black",
                  "fontSize": 20,
                  "fontWeight": "bold",
                },
              ]
            }
          >
            Sign Up
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderColor": "#8BF0BC",
              "borderRadius": 10,
              "borderWidth": 1,
              "marginBottom": 20,
              "opacity": 1,
              "padding": 12,
              "width": "90%",
            }
          }
          testID="authButton"
        >
          <Text
            style={
              [
                {
                  "fontFamily": "Quicksand",
                },
                {
                  "color": "black",
                  "fontSize": 20,
                  "fontWeight": "bold",
                },
              ]
            }
          >
            Back
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
